translators/src/Tests.hs

```diff
@@ -7,301 +7,234 @@ import GHC.Natural (minusNatural)
 import Grammar
 
 -- re-usable generators for the tests below
-genIndexName :: Char -> Natural -> String
-genIndexName c i = c : show i
 
-genIndex :: Char -> Natural -> [String]
-genIndex c m = map (genIndexName c) [1..m]
+-- the 'n == 0' case is often trivial
+trivial :: Natural -> [a] -> [a]
+trivial n c = if n == 0 then [] else c
+
+-- there is likely a nicer way to say this?
+-- (The code is 1-based, keep that for now)
+iter :: Natural -> (Natural -> a) -> [a]
+iter n f = map f [1..n]
 
-optional :: Natural -> [a] -> [a]
-optional n c = if n == 0 then [] else c
+-- various name and variable generators
+nm :: Char -> Natural -> String
+nm c n = c : show n
+
+genIndex :: Char -> Natural -> [String]
+genIndex c m = iter m (nm c)
 
 v :: Char -> Natural -> Expr
-v c n = Var $ c : show n
+v c n = Var $ nm c n
 
 vx :: Natural -> Expr
 vx = v 'x'
 
 xs :: Natural -> Expr
```

Comment:
Rename to `vxs`? Let's get rid of this eventually, but it is at least consistent with `vx`.